### PR TITLE
RPDE

### PR DESCRIPTION
**Pull request by ylt**

### CODE REVIEW

File: lib/openactive/rpde/rpde_body.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      def self.create_from_next_change_number(feed_base_url, change_number, items)
+        rpde_body = self.new(
+          items: items
+        )
+        if items.length > 0
+          first_item = items.first
+
+          # Checks that the afterId and afterTimestamp provided are not the
+          # first item in the feed (helps detect whether query is correct)
+          if first_item.modified == change_number
```
Comment (henryaddison) [inline lib/openactive/rpde/rpde_body.rb:112]: `<=`?

File: lib/openactive/rpde/rpde_body.rb [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      def self.create_from_modified_id(feed_base_url, modified, id, items)
+        rpde_body = self.new(
+          items: items
+        )
+        if items.length > 0
+          first_item = items.first
+
+          # Checks that the afterId and afterTimestamp provided are not the
+          # first item in the feed (helps detect whether query is correct)
+          if first_item.modified.to_s == modified.to_s && first_item.id.to_s == id.to_s
```
Comment (henryaddison) [inline lib/openactive/rpde/rpde_body.rb:57]: `(first_item.modified.to_s < modified.to_s) || (first_item.modified.to_s == modified.to_s && first_item.id.to_s <= id.to_s)` maybe?

File: lib/openactive/concerns/json_ld_serializable.rb [hunk 3]
```diff
@@ -10,10 +10,12 @@ def properties
           @attributes ||= begin
             if superclass.respond_to?(:properties)
               superclass.properties.clone
-            else
+            elsif method_defined?(:type)
```
Comment (henryaddison) [inline lib/openactive/concerns/json_ld_serializable.rb]: `responds_to?` instead maybe?

**Review by henryaddison — APPROVED** [review]
Looks good. Three comments inline but I leave it entirely to your judgement if any are worth acting on.

File: lib/openactive/rpde/rpde_body.rb (hunk 1 above)
Comment (ylt) [reply lib/openactive/rpde/rpde_body.rb:112]: In it's current state it's identical to the PHP and .NET implementations:
- https://github.com/openactive/models-php/blob/master/src/Rpde/RpdeBody.php#L203-L207
- https://github.com/openactive/OpenActive.NET/blob/master/OpenActive.NET/rpde/RDPEModels.cs#L210

File: lib/openactive/rpde/rpde_body.rb (hunk 2 above)
Comment (ylt) [reply lib/openactive/rpde/rpde_body.rb:57]: Wouldn't work due to being strings now, would need to attempt to parse them.

Also, in it's current state it'sa identical to the PHP and .NET implementations:
- https://github.com/openactive/models-php/blob/master/src/Rpde/RpdeBody.php#L99-L104
 - https://github.com/openactive/OpenActive.NET/blob/master/OpenActive.NET/rpde/RDPEModels.cs#L148-L150

**Pull request merged**